kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/DeltaLogFile.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.replay;
+
+import io.delta.kernel.utils.FileStatus;
+import io.delta.kernel.internal.checkpoints.CheckpointInstance;
+import io.delta.kernel.internal.fs.Path;
+import io.delta.kernel.internal.util.FileNames;
+
+/** Internal wrapper class holding information needed to perform log replay. Represents either a
+ * Delta commit file, classic checkpoint, a multipart checkpoint, or a sidecar checkpoint.
```

Comment:
This class may also represent the v2 checkpoint top-level file? Add the same to the list.